### PR TITLE
feat(notifications): Implement GET /notification/id/{id} V2 API

### LINKED ISSUE #3207: [V2 notifications] Implement GET /notification/id/{id} V2 API
# 🚀 Feature Request

### Relevant Package

 This feature request is for support-notifications.

### Description
Per [https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/default/get_notification_id__id_](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/default/get_notification_id__id_), implement GET /notification/id/{id} V2 API.

### PR DESCRIPTION
**Pull request by FelixTing**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the new behavior?
Per [https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/default/get_notification_id__id_](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/default/get_notification_id__id_), implemented GET /notification/id/{id} V2 API.

Close #3207

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by FelixTing** [comment]
recheck

File: internal/support/notifications/v2/controller/http/notification_test.go [hunk 1]
```diff
@@ -39,7 +41,7 @@ var (
 	testNotificationSeverity    = models.Normal
 )
 
-var testAddNotificationRequest = dtos.Notification{
+var testAddNotification = dtos.Notification{
```
Comment (lenny-intel) [inline internal/support/notifications/v2/controller/http/notification_test.go:44]: Why does `Notification` not have a factory method to use here?

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/support/notifications/v2/controller/http/notification_test.go (hunk 1 above)
Comment (FelixTing) [reply internal/support/notifications/v2/controller/http/notification_test.go]: Modified. Use the factory method to create a `Notification` DTO for testing.

**Comment by FelixTing** [comment]
rebased

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**